bluemira/power_cycle/loads.py

```diff
@@ -0,0 +1,1065 @@
+        ----------
+        ax: Axes
+            Instance of the `matplotlib.axes.Axes` class. By default,
+            the currently selected axes are used.
+        **kwargs: dict
+            Options for the `scatter` method.
+
+        Returns
+        -------
+        plot_list: list
+            List of plot objects created by the `matplotlib` package.
+            The first element of the list is the plot object created
+            using the `pyplot.scatter`, while the second element of the
+            list is the plot object created using the `pyplot.text`
+            method.
+        """
+
+        # Validate axes
+        ax = PowerCycleUtilities.validate_axes(ax)
+
+        # Retrieve default plot options
+        default = self._scatter_kwargs
+
+        # Set each default options in kwargs, if not specified
+        kwargs = PowerCycleUtilities.add_dict_entries(kwargs, default)
+
+        # Retrieve instance characteristics
+        name = self.name
+        time = self.time
+        data = self.data
+
+        # Preallocate output
+        plot_list = []
+
+        # Plot
+        label = name + " (data)"
+        plot_obj = ax.scatter(time, data, label=label, **kwargs)
+        plot_list.append(plot_obj)
+
+        # Add text to plot
+        index = self._text_index
+        text = f"{name} (PowerData)"
+        label = name + " (name)"
+        angle = self._text_angle
+        plot_obj = ax.text(time[index], data[index], text, label=label, rotation=angle)
+        plot_list.append(plot_obj)
+
+        # Return list of plot objects
+        return plot_list
+
+
+# ######################################################################
+# GENERIC POWER LOAD
+# ######################################################################
+class PowerLoad(PowerCycleABC):
+    """
+    Generic representation of a power load.
+
+    Defines a power load with a set of `PowerData` instances. Each
+    instance must be accompanied by a `model` specification, used to
+    compute additional values between data points. This enables the
+    instance to compute time-dependent curves.
+
+    Parameters
+    ----------
+    name: str
+        Description of the `PowerLoad` instance.
+    data_set: PowerData | list[PowerData]]
+        Collection of instances of the `PowerData` class that define
+        the `PowerLoad` object.
+    model: str | list[str]
+        List of types of model that defines values between points
+        defined in the `load` Attribute. Valid models include:
+
+        - 'ramp';
+        - 'step';
+
+        that are defined in the `_single_curve` method.
+    """
+
+    # ------------------------------------------------------------------
+    # CLASS ATTRIBUTES
+    # ------------------------------------------------------------------
+
+    # Default number of points in each curve segment
+    _n_points = 100
+
+    # Implemented models (add model name here after implementation)
+    _valid_models = ["ramp", "step"]
```

Comment:
What did you have in mind? A small 'model' class with a list of models, that can be expanded, each item associated to a function/method for computing the interpolation?

Reply:
It would be quite simple:
```python
from enum import Enum, auto

class PowerCycleModel(Enum):
    RAMP = auto()
    STEP = auto()
```

Its usually convention for `Enum` variable names to be captialised . An `Enum` will automatically add existence checking with no extra work from the developer. Have a read through @hsaunders1904 's link.